### PR TITLE
Bug 1206892 - null entries in init.json break the world r=Cwiiis

### PR DESCRIPTION
**Pull request by fabricedesre**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=9d8e4984c81c5638f5add32569c0df3affaa163d

File: apps/homescreen/js/firstrun.js [hunk 1]
```diff
@@ -24,7 +24,7 @@
             var index = 0;
             json.grid.forEach((section) => {
               section.forEach((entry) => {
-                if (!entry.manifestURL) {
+                if (!entry || !entry.manifestURL) {
```
Comment (Cwiiis) [inline apps/homescreen/js/firstrun.js]: We should unit test this, it already unit tests for null and empty JSON, but it doesn't test for null entries.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=d9806b5fc1873a7dbd2f161d038322628afd1dac

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=d541e9005447ea9ca5b2d5c55d804c0a07856e52

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=e16121162f94906b3d17462f45aa7977a3db0fd0

**Pull request merged**